File: samples/CslaNet/cs/DeepData/www/DeepDataService/App_Code/DeepDataDAL.cs
using System.Web.Services;
using Csla.Data;
using DeepData.DAL;
using DeepData.DTO;

[WebService(Namespace = "http://ws.lhotka.net/")]
[WebServiceBinding(ConformsTo = WsiProfiles.BasicProfile1_1)]
public class DeepDataDAL : WebService
{
	[WebMethod]
	public DeepData.DTO.OrderDto[] GetOrders()
	{
		var df = new DataFactory();
		using (var dal = df.GetOrderDataObject())
		{
			return ((DeepData.DTO.OrderDto[])(dal.GetOrders()));
		}
	}
}

File: Samples/CslaNet/vb/DeepData/www/DeepDataService/App_Code/DeepDataDAL.vb
Option Strict On

Imports System.Collections.Generic
Imports System.Web
Imports System.Web.Services
Imports System.Web.Services.Protocols

<WebService(Namespace:="http://ws.lhotka.net/")> _
<WebServiceBinding(ConformsTo:=WsiProfiles.BasicProfile1_1)> _
<Global.Microsoft.VisualBasic.CompilerServices.DesignerGenerated()> _
Public Class DeepDataDAL
  Inherits System.Web.Services.WebService

  <WebMethod()> _
  Public Function GetOrders() As DeepData.DTO.OrderDto()

    Dim df As New DeepData.DAL.DataFactory
    Using dal As DeepData.DAL.OrderData = df.GetOrderDataObject
      Return CType(dal.GetOrders, DeepData.DTO.OrderDto())
    End Using

  End Function

End Class
